### PR TITLE
destroy existing infrastructure

### PR DESCRIPTION
**Pull request by razzkumar**

### CODE REVIEW

File: cli/cmd/destroy.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package cmd
+
+import "github.com/leapfrogtechnology/shift/cli/internals/destroy"
+
+// Destroy delete existing infrastructure
```
Comment (pratishshr) [inline cli/cmd/destroy.go]: deletes*

File: cli/internals/destroy/destroy.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+	"github.com/leapfrogtechnology/shift/core/services/slack"
+	"github.com/leapfrogtechnology/shift/core/services/storage"
+	"github.com/leapfrogtechnology/shift/core/utils/file"
+	"github.com/leapfrogtechnology/shift/core/utils/system"
+	"github.com/leapfrogtechnology/shift/core/utils/system/exit"
+	"github.com/leapfrogtechnology/shift/infrastructure/internals/terraform"
+)
+
+// Run initializes new environment.
```
Comment (pratishshr) [inline cli/internals/destroy/destroy.go]: destroys*

File: cli/internals/destroy/destroy.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		slack.Notify(project.SlackURL, fmt.Sprintf("Error :\n %s", err.Error()), "#04EBB8")
+		os.Exit(1)
+
+	} else {
+		err := terraform.MakeTempAndDestroy(project, environment, workspaceDir)
+		slack.Notify(project.SlackURL, fmt.Sprintf("Error :\n %s", err.Error()), "#04EBB8")
+		os.Exit(1)
+	}
+
+	slack.Notify(project.SlackURL, fmt.Sprintf(" 🎉 🎉 🎉Successfully destroyed *%s* env of *%s* from *%s*. 🎉 🎉 🎉", environment, project.Name, project.Platform), "#04EBB8")
```
Comment (pratishshr) [inline cli/internals/destroy/destroy.go]: let's not send slack notifications when we are destroying the environment

File: core/utils/file/file.go [hunk 4]
```diff
@@ -17,3 +18,11 @@ func GetFileContentType(filename string) string {
 
 	return mime.TypeByExtension("." + ext)
 }
+
+// IsExist check existence of file
+func IsExist(filepath string) bool {
```
Comment (pratishshr) [inline core/utils/file/file.go]: Let's call this function `Exists` instead of `IsExist`

File: infrastructure/internals/terraform/terraform.go [hunk 5]
```diff
@@ -112,3 +117,43 @@ func RunInfrastructureChanges(workspaceDir string, workspaceName string) (string
...
+func DestroyInfrastructure(workspaceDir string) error {
+	fmt.Println("Distroying Infrastructure...")
+	s := spinner.New(spinner.CharSets[11], 100*time.Millisecond) // Build our new spinner
+	s.Prefix = "  "
+	_ = s.Color("cyan", "bold")
+	s.Start()
+	cmd := exec.Command("terraform", "destroy", "--auto-approve")
+	cmd.Dir = workspaceDir
+	// var stdout bytes.Buffer
+	// var stderr bytes.Buffer
```
Comment (pratishshr) [inline infrastructure/internals/terraform/terraform.go]: Remove unwanted code

File: infrastructure/internals/terraform/terraform.go [hunk 6]
```diff
@@ -112,3 +117,43 @@ func RunInfrastructureChanges(workspaceDir string, workspaceName string) (string
 
 	return out, err
 }
+
+// DestroyInfrastructure destroy existing infrastructure
+func DestroyInfrastructure(workspaceDir string) error {
+	fmt.Println("Distroying Infrastructure...")
+	s := spinner.New(spinner.CharSets[11], 100*time.Millisecond) // Build our new spinner
```
Comment (pratishshr) [inline infrastructure/internals/terraform/terraform.go]: Do we need to create a new spinner? 
There's a spinner in the utility as well

Comment (razzkumar) [reply infrastructure/internals/terraform/terraform.go]: I have tried but unable to import due to other spinner module that is already imported but now i will modify it and use it

File: cli/internals/destroy/destroy.go [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+	workspaceDir := filepath.Join(workspaceRoot, project.Name, project.Type, environment)
+
+	terraformFile := workspaceDir + "/infrastructure.tf"
+
+	isExist := file.Exists(terraformFile)
+
+	if isExist {
+		terraform.DestroyInfrastructure(workspaceDir)
+
```
Comment (pratishshr) [inline cli/internals/destroy/destroy.go]: remove unwanted line

File: cli/internals/destroy/destroy.go [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+		exit.Error(errors.New("Unknown Environment type "+"'"+environment+"'"), "Error")
+	}
+
+	workspaceRoot := "/tmp"
+
+	workspaceDir := filepath.Join(workspaceRoot, project.Name, project.Type, environment)
+
+	terraformFile := workspaceDir + "/infrastructure.tf"
+
+	isExist := file.Exists(terraformFile)
```
Comment (pratishshr) [inline cli/internals/destroy/destroy.go]: name this variable as `exists` as well

File: infrastructure/internals/terraform/terraform.go [hunk 9]
```diff
@@ -112,3 +111,39 @@ func RunInfrastructureChanges(workspaceDir string, workspaceName string) (string
...
+	message := " ERROR While Destroying Infrastructure"
+	if err != nil {
+		logger.LogError(err, message)
+		spinner.Stop()
+		return errors.New(err.Error() + message)
+	}
+
+	spinner.Stop()
+	return nil
+
```
Comment (pratishshr) [inline infrastructure/internals/terraform/terraform.go]: remove unwanted line here

**Pull request merged**